### DISCUSSION TITLE
Machine Learning Tutorial?

### DISCUSSION THREAD

**Discussion by finickyFalcon5** [immich-app/immich#807] [Q&A]
Hi,

Is there a tutorial available for implementing Machine Learning within Immich?  Do I need to install anything? How do I run it? How do I teach it to identify objects/people?

Thanks.

**Comment by alextran1502** [comment]
Hello,
Right now the machine learning is operating on its own. Object detection and image classification run automatically when you upload a new photo or video. 

There isn't a built in teaching mechanism for a re-train right now,

**Reply by florihupf** [reply]
Hmm,

How do I know it works? Am I supposed to see those tags somewhere?  I tried to find some info, but no dice.

Thanks for any pointers!

**Reply by alextran1502** [reply]
@florihupf in the search page on the mobile app. you should see the section "Things" where all the result are listed as group. You can also search those tags in the search bar. We are working on a revamp of the tag system and putting it together with the search function on the web

**Reply by florihupf** [reply]
@alextran1502 D'oh! :- So obvious! Great, thank you! :- Also thanks for all the great work, it is really awesome!
